src/main/scala/firrtl/passes/Inline.scala

```diff
@@ -353,8 +358,11 @@ class InlineInstances extends Transform with DependencyAPIMigration with Registe
         Some(m.map(onStmt(ModuleName(m.name, CircuitName(c.main)))))
     })
 
-    val renames = renamesSeq.tail.foldLeft(renamesSeq.head)(_ andThen _)
+    val renames = renamesSeq match {
+      case Nil        => None
+      case car :: cdr => Some(cdr.foldLeft(car)(_ andThen _))
```

Comment:
```suggestion
      case car +: cdr => Some(cdr.foldLeft(car)(_ andThen _))
```
It's not actually guaranteed that `renamesSeq` is a `List` so if someone were to tweak the code making the Seq, this could become a runtime exception

Reply:
The `reduceLeftOption` is clearly cleaner than what I wrote here (with the enormous detriment of inability to include car/cdr lisp-isms...). That said, I do think it's an anti-pattern to defensively program for the reason of heading off future, incorrect modifications an internal API (and a method-internal API at that). In that case, it's probably safer to specify the actual state the type of `renamesSeq` and make sure that there's sufficient coverage on the tests to hit the zero-, one-, and two-element cases.

Reply:
I don't agree it's an antipattern because it is technically incorrect to use List matching on a Seq. List pattern matching is correct by coincidence, not by API.

Reply:
Didn't know. Yeah, this is a real problem. Thanks for catching this. :+1:

How is this actually working in the original I wrote? It's matching on the [case class `::`](https://www.scala-lang.org/api/current/scala/collection/immutable/$colon$colon.html), but how does the `Seq` allowed to match a type of `List`? I mean, you can't pass a `Seq[Int]` into `def foo(a: List[Int])`, but you can pattern pattern match it. I guess this stems from the contravariance of `PartialFunction[-A, +B]`?

Reply:
So technically it is okay to pattern match a trait with concrete subclasses because maybe you want to do something different for each one, eg.
```scala
someSeq match {
  case head :: tail => // Do something List specific
  case vec: Vector[_] => // Do something Vector specific
  ...
}
```
The `object Seq` factory methods return `List` as their default type, but this is not technically the API even though it will probably never change because it would break tons of code that assumes it.

This kind of thing *should* give a match exhaustiveness check warning, but that only works on case class-generated unapply in Scala 2. Scala 3 does support checking this on custom unapply, so maybe this will be better then.